sdk/core/core/src/az_span.c

```diff
@@ -29,16 +29,16 @@ AZ_NODISCARD az_span az_span_from_str(char* str)
 {
   AZ_PRECONDITION_NOT_NULL(str);
 
-  int32_t length = (int32_t)strlen(str);
-  return az_span_init((uint8_t*)str, length, length);
+  int32_t const length = (int32_t)strlen(str);
+  return az_span_init((uint8_t*)str, length, length + 1);
 }
 
 AZ_NODISCARD az_span az_span_slice(az_span span, int32_t low_index, int32_t high_index)
 {
   AZ_PRECONDITION_RANGE(-1, high_index, az_span_capacity(span));
   AZ_PRECONDITION(high_index == -1 || (high_index >= 0 && low_index <= high_index));
 
-  int32_t capacity = az_span_capacity(span);
+  int32_t const capacity = az_span_length(span);
```

Comment:
Why did you change this to length? We want capacity here.

Reply:
I will undo this change, sorry that it got here. I was trying to see how hard would it be to change `AZ_SPAN_FROM_STR()` in a way that for `"Hello"` it has `length = 5` but `capacity = 6`.

Reply:
Undone.